### PR TITLE
YANG Validation for MCLAG, NAT, MUXCABLE tables

### PR DESCRIPTION
**Pull request by isabelmsft**
#### What I did
Add YANG validation using GCU for writes to MCLAG, NAT, MUXCABLE tables tables in ConfigDB
#### How I did it
Using same method as https://github.com/sonic-net/sonic-utilities/pull/2190/files, extend to  MCLAG, NAT, MUXCABLE tables
#### How to verify it
verified testing on virtual switch CLI

### CODE REVIEW

**Comment by qiluo-msft** [comment]
Please add this PR to your HLD PR's description code PR list, so people can find relevant HLD.

File: config/mclag.py [hunk 1]
```diff
@@ -121,34 +124,42 @@ def mclag(ctx):
...
         if domain_id in mclag_domain_keys:
-            db.mod_entry('MCLAG_DOMAIN', domain_id, fvs)
-        else: 
-            ctx.fail("only one mclag Domain can be configured. Already one domain {} configured ".format(mclag_domain_keys[0]))  
+            try:
+                db.mod_entry('MCLAG_DOMAIN', domain_id, fvs)
+            except ValueError as e:
+                ctx.fail("Invalid ConfigDB. Error: {}".format(e))
+        else:
+            ctx.fail("only one mclag Domain can be configured. Already one domain {} configured ".format(list(mclag_domain_keys)[0]))
```
Comment (qiluo-msft) [inline config/mclag.py:162]: >list [](http://example.com/codeflow?start=105&length=4)

Just wondering whether this is a bug fix?

File: config/nat.py [hunk 2]
```diff
@@ -711,10 +769,13 @@ def add_pool(ctx, pool_name, global_ip_range, global_port_range):
...
     if entryFound == False:
-        config_db.set_entry(table, key, {dataKey1: global_ip_range, dataKey2 : global_port_range})
+        try:
+            config_db.set_entry(table, key, {dataKey1: global_ip_range, dataKey2 : global_port_range})
+        except ValueError as e:
+            ctx.fail("Invalid ConfigDB. Error: {}".format(e))
 
 #
-# 'nat add binding' command ('config nat add binding <binding_name> <pool_name> <acl_name>')
+# 'nat add binding' command ('config nat add binding <ninding_name> <pool_name> <acl_name>')
```
Comment (qiluo-msft) [inline config/nat.py:778]: >ninding_name [](http://example.com/codeflow?start=54&length=12)

typo?

File: config/console.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import click
+import jsonpatch
```
Comment (ganglyu) [inline config/console.py:2]: Please use "from jsonpatch import xxx" instead

File: config/console.py [hunk 4]
```diff
@@ -83,15 +96,18 @@ def add_console_setting(db, linenum, baud, flowcontrol, devicename):
...
+    ctx = click.get_current_context()
 
     table = "CONSOLE_PORT"
 
     data = config_db.get_entry(table, linenum)
     if data:
-        config_db.mod_entry(table, linenum, None)
+        try:
+            config_db.set_entry(table, linenum, None)
+        except JsonPatchConflict as e:
```
Comment (ganglyu) [inline config/console.py:108]: There're two exceptions: ValueError and JsonPatchConflict.
Why do you use JsonPatchConflict here and use ValueError for others?

File: config/mclag.py [hunk 5]
```diff
@@ -1,8 +1,12 @@
 
 import click
 from swsscommon.swsscommon import ConfigDBConnector
+from .validated_config_db_connector import ValidatedConfigDBConnector
 import ipaddress
-
+import jsonpatch
+from jsonpatch import JsonPatchConflict
```
Comment (ganglyu) [inline config/mclag.py:7]: Can you use the same import method?
"import jsonpatch" or "from jsonpatch import xxx"

File: config/nat.py [hunk 6]
```diff
@@ -1,8 +1,14 @@
 import ipaddress
 
 import click
+import jsonpatch
+import jsonpointer
+from jsonpatch import JsonPatchConflict
+from jsonpointer import JsonPointerException
```
Comment (ganglyu) [inline config/nat.py:7]: Please use the same import method

File: config/nat.py [hunk 7]
```diff
@@ -1,8 +1,14 @@
...
 from swsscommon.swsscommon import SonicV2Connector, ConfigDBConnector
+from .validated_config_db_connector import ValidatedConfigDBConnector
 
+ADHOC_VALIDATION = True
```
Comment (ganglyu) [inline config/nat.py:11]: Every py file update config has defined ADHOC_VALIDATION, is there any conflict?

File: config/mclag.py (hunk 1 above)
Comment (isabelmsft) [reply config/mclag.py:162]: Yes. Without the change, we get this error
```
  File "/usr/local/lib/python3.9/dist-packages/config/mclag.py", line 161, in add_mclag_domain
    print(mclag_domain_keys[0])
TypeError: 'dict_keys' object is not subscriptable 
```

File: config/nat.py (hunk 7 above)
Comment (isabelmsft) [reply config/nat.py:11]: There is no conflict, as the variable scope is limited to the file

File: config/nat.py (hunk 2 above)
Comment (isabelmsft) [reply config/nat.py]: yes, corrected the typo now

File: config/console.py (hunk 3 above)
Comment (isabelmsft) [reply config/console.py]: cleaned import statement

File: config/console.py (hunk 4 above)
Comment (isabelmsft) [reply config/console.py:107]: ValueError is thrown when attempting to set a field that already exists to a value that the Yang model does not allow. 

JsonPatchConflict is thrown when attempting to remove, or set a value to None, when the field does not already exist.

File: config/mclag.py (hunk 5 above)
Comment (isabelmsft) [reply config/mclag.py:6]: cleaned import statement

File: config/nat.py (hunk 6 above)
Comment (isabelmsft) [reply config/nat.py:5]: cleaned import statements

**Review by ganglyu — APPROVED** [review]

File: config/mclag.py (hunk 1 above)
Comment (qiluo-msft) [reply config/mclag.py:162]: Thanks for your contribution to fix a bug in codebase!

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**